src/events/deposits.upsert.service.ts

```diff
@@ -288,4 +228,190 @@ export class DepositsUpsertService {
+                  transaction_hash: deposit.transaction_hash,
+                  graffiti: deposit.graffiti,
+                })),
+                network_version: networkVersion,
+              },
+            });
+
+            // Now create new not existing deposits
+            await prisma.deposit.createMany({
+              data: depositParams,
+              skipDuplicates: true,
+            });
+
+            deposits = await prisma.deposit.findMany({
+              where: {
+                block_hash: blockHash,
+                network_version: networkVersion,
+              },
+            });
+
+            const eventPayloads = [];
+            const usersFiltered = new Map<string, User>();
+
+            // Create the event payloads filtering events and users on points < 0
+            for (const deposit of deposits) {
+              const points = this.eventsService.calculateDepositPoints(deposit);
+
+              if (points <= 0) {
+                continue;
+              }
+
+              const user = users.get(deposit.graffiti);
+
+              // This should NEVER happen but can happen if a user is deleted after they made a deposit
+              if (!user) {
+                continue;
+              }
+
+              usersFiltered.set(deposit.graffiti, user);
+
+              eventPayloads.push({
+                occurred_at: blockTimestamp.toISOString(),
+                type: EventType.SEND_TRANSACTION,
+                user_id: user.id,
+                points: points,
+                deposit_id: deposit.id,
+              });
+            }
+
+            users = usersFiltered;
+
+            // This SHOULD NOT be needed but this cleans up bad data in the server and crashes without this
+            await prisma.event.deleteMany({
+              where: {
+                deposit_id: {
+                  in: deposits.map((deposit) => deposit.id),
+                },
+              },
+            });
+
+            await prisma.event.createMany({
+              data: eventPayloads,
+            });
+          } else if (operation.type === BlockOperation.DISCONNECTED) {
+            await prisma.deposit.updateMany({
+              data: {
+                main: false,
+              },
+              where: {
+                block_hash: blockHash,
+                network_version: networkVersion,
+              },
+            });
+
+            deposits = await prisma.deposit.findMany({
+              where: {
+                block_hash: blockHash,
+                network_version: networkVersion,
+              },
+            });
+
+            await prisma.event.deleteMany({
+              where: {
+                deposit_id: {
+                  in: deposits.map((deposit) => deposit.id),
+                },
+              },
+            });
+
+            users = await this.usersService.findManyAndMapByGraffiti(
+              deposits.map((d) => d.graffiti),
```

Comment:
nit - two loops over deposits here and on 384, above

Reply:
I don't see that, do you mean the one in the connected branch?

Reply:
The one up above in `deleteMany` to map over deposits to get the IDs